modules/apps/27-interchain-accounts/types/metadata.go

```diff
@@ -169,6 +169,9 @@ func validateConnectionParams(metadata Metadata, controllerConnectionID, hostCon
 	}
 
 	if metadata.HostConnectionId != hostConnectionID {
+		if metadata.HostConnectionId == "" {
+			return nil
```

Comment:
This is not how we should do it. It is possible that we want to add other checks after this in the future. So, the only `nil` returned should be at the end of the function, not inside this if statement

Reply:
I think we need to be a bit carefully with the validation of `metadata.HostConnectionId` because it will be valid to be empty in `ChanOpenInit` but it should not be empty anymore in `ChanOpenAck`, right?

Reply:
I'd make the following suggestion @GNaD13 :
1. Revert your changes to validateConnectionParams.
2. In `ChanOpenTry`, before sending the metadata to validation, override `metadata.HostConnectionId` to be `connectionHops[0]` regardless of whether or not it is filled.
3. Send to validation after step 2.

Note that this is okay because the host is allowed to propose a new version string in `ChanOpenTry`. Does this sound good @crodriguezvega?

Reply:
If so, I think we will also remove the validation in [ChanOpenInit](https://github.com/cosmos/ibc-go/blob/dc99ee43c1ade80094d56acd00ca16097b308d44/modules/apps/27-interchain-accounts/controller/keeper/handshake.go#L64)